src/hostnet/lib/slirp.ml

```diff
@@ -152,9 +156,9 @@ let connect x peer_ip local_ip =
             );
             Tcpip_stack.listen_tcpv4_flow s ~on_flow_arrival:(
               fun ~src:(src_ip, src_port) ~dst:(dst_ip, dst_port) ->
-
                 let for_us src_ip = Ipaddr.V4.compare src_ip local_ip == 0 in
-                ( if for_us src_ip && src_port = 53 then begin
+                let for_dns src_ip = for_us src_ip || Ipaddr.V4.compare src_ip extra_dns_ip == 0 in
+                ( if for_dns src_ip && src_port = 53 then begin
                     Resolv_conf.get () (* re-read /etc/resolv.conf *)
                     >>= function
                     | (Ipaddr.V4 ip, port) :: _  -> Lwt.return (ip, port)
```

Comment:
Should we send the TCP traffic sent to the `extra_dns_ip` to the secondary DNS server as well as the UDP?